backend/imap/conn.go

```diff
@@ -151,7 +151,7 @@ func (b *conns) idle(clt *client) error {
 	c := clt.conn
 
 	mailbox := "INBOX"
-	if c.Mailbox != nil && c.Mailbox.Name != mailbox {
+	if c.Mailbox != nil && c.Mailbox().Name != mailbox {
```

Comment:
`c.Mailbox` is never `nil` because it's a function. You want to check if `c.Mailbox()` is `nil` instead.